cedar-policy/src/api.rs

```diff
@@ -2338,6 +2346,7 @@ pub fn eval_expression(
 }
 
 #[cfg(test)]
+#[cfg(feature = "partial-eval")]
 mod test {
```

Comment:
could we maybe rename the `test` module to be clear it's only tests related to partial evaluation?  Doesn't have to block this PR though